compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -360,12 +367,33 @@ def process_pending_tasks():
                     if self.time_to_quit:
                         self.stop()
                         continue  # nominally == break; but let event do it
+
                     try:
-                        incoming_task = self.pending_task_queue.get(timeout=1)
-                        task_msg = unpack(incoming_task)
+                        prop_headers, body = self.pending_task_queue.get(timeout=1)
+                        task_msg = unpack(body)
                         if not isinstance(task_msg, Task):
                             raise InvalidMessageError()
 
```

Comment:
In the interest of not doing extraneous work, you might consider moving this stanza to below the check.  That is, in the case that it's a disallowed function, why go through unpacking it at all?

Reply:
Yes it was originally below the check.  The reason I moved it above was a combination of "invalid message or disallowed func are both FailNow, so doesn't matter the order" and also getting the task_id out of the unpacked task_msg was slightly better than getting the one from the prop_header.

But you are right in that checking allowed_func is light weight, and should be done first compared to the variable-chunk-size unpack.  Let me rearrange it.

Reply:
Rearranging was a bit more complicated, as without unpacking the task_msg we don't know what the task_id is, and headers have to be manually set in tests.  Thus now the slightly redundant log.exception for missing headers and additional InvalidMessageError if tid is missing but fid is present.  Still, I left task_msg unpacked til later.